### PR TITLE
{chem}[foss/2019a] PSI4 v1.3.1 w/ Python 3.7.2

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
~#8355~ (PCMSolver), ~#8356~ (CheMPS2), ~~#8361~~ (libxc w/ CMake)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2611.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2cb93ef91a63f4ef133301b6c4777729 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3166.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/a3528a367362cb981d7b8c68da9d03a0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2071.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/54c87ef30094a215efc44b0a3d1afbc6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2400.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/38f5ab5329d831473c973efe3b29f7bb for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/892af28f1415c2f656e2afba6fb3b526 for a full test report.

**Comment by boegel** [comment]
@vanzod looks like #8361 is missing, since `PSI4` is not picking up the provided `libxc` but trying to build the (older) internal version (which fails because of https://gitlab.com/libxc/libxc/issues/93)

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 5 out of 6 (1 easyconfigs in this PR)
nic400 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/6893b885a0e909eee7733fed8f975ded for a full test report.

**Comment by boegel** [comment]
@wpoely86 Actual error is missing from log in test report, any ideas what went wrong?

**Comment by wpoely86** [comment]
```
libxc.a(gga_c_am05.c.o):gga_c_am05.c:function xc_gga_c_am05_func: error: undefined reference to 'log'
```

A missing `-lm` it seems.

**Comment by wpoely86** [comment]
a no, it's an old `libxc`, will reinstall

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3121.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/98fdad08b4d352720c16f6b55a2e7fb8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2022.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a0bf77dfb94c1ab6e299c18c42a297aa for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 7 out of 7 (1 easyconfigs in this PR)
nic166 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/190da9e5c32c07f7bf32a2b8cb688ee7 for a full test report.

File: easybuild/easyconfigs/p/PSI4/PSI4-1.3.1_fix_cmake_release.patch [hunk 1]
```diff
@@ -0,0 +1,95 @@
+diff -ru psi4-1.3.1.orig/cmake/autocmake_safeguards.cmake psi4-1.3.1/cmake/autocmake_safeguards.cmake
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/PSI4/PSI4-1.3.1_fix_cmake_release.patch]: purpose and author?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2499.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/df3643657f0f8cc4c8f4a73b20b40fdb for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/a5de030d762d95fed6d8485dc734a8cd for a full test report.

**Comment by wpoely86** [comment]
@akesandgren did you use https://github.com/easybuilders/easybuild-easyconfigs/pull/8361 ?

**Comment by akesandgren** [comment]
Yes I did

**Comment by boegel** [comment]
@akesandgren That's positively weird...

Your `Python` 3.7.2 installation is complete, right? It does have `setuptools` installed?

Do you have another `python3` on this system that it's picking up on?

**Comment by akesandgren** [comment]
It should have...
the module says that setuptools-40.8.0 is included

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/4564558f6394b85e9309d620672b6f02 for a full test report.

**Comment by boegel** [comment]
@akesandgren Can you easily reproduce the problem you saw earlier?

**Comment by akesandgren** [comment]
Trying, test report should arrive later...

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/81e43c081cdfebc737cb4ce84bb9fd13 for a full test report.

**Comment by boegel** [comment]
@akesandgren Partial log doesn't show the actual problem? Do you mind sharing a full debug log for this?

**Comment by boegel** [comment]
(re-triggering Travis checks)

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-cn0823.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/dc83edb9c0f9a91962bddf2108983d45 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-cn0823.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/da41c1cfdd6f55d05025d3d057f64a27 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-cn0823.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/43b2f04a3e588cad3d66f233d885676d for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
b-cn0823.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/5254fb3a5e4c91a7c05e76c68d2335b5 for a full test report.

**Comment by akesandgren** [comment]
The actual problem could be this:
```
cd /scratch/eb-ake/PSI4/1.3.1/foss-2019a-Python-3.7.2/obj/external/upstream/qcelemental/q
celemental_external-prefix/src/qcelemental_external && /usr/bin/python3.5 setup.py build
Traceback (most recent call last):
  File "setup.py", line 3, in <module>
    import setuptools
ImportError: No module named 'setuptools'
external/upstream/qcelemental/CMakeFiles/qcelemental_external.dir/build.make:114: recipe for target 'external/upstream/qcelemental/qcelemental_external-prefix/src/qcelemental_external-stamp/qcelemental_external-build' failed
make[2]: *** [external/upstream/qcelemental/qcelemental_external-prefix/src/qcelemental_external-stamp/qcelemental_external-build] Error 1
make[2]: Leaving directory '/scratch/eb-ake/PSI4/1.3.1/foss-2019a-Python-3.7.2/obj'
CMakeFiles/Makefile2:1561: recipe for target 'external/upstream/qcelemental/CMakeFiles/qcelemental_external.dir/all' failed
make[1]: *** [external/upstream/qcelemental/CMakeFiles/qcelemental_external.dir/all] Error 2
```

Note that it uses the OS installed python3.5

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 7 out of 7 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/6e6bb5ded75fdfa3bbe0b995e0074f0f for a full test report.

**Comment by boegel** [comment]
@akesandgren Can you retry this with current `develop`? I hope your changes in https://github.com/easybuilders/easybuild-easyblocks/pull/2141 fix the problem with the wrong Python being used?

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/92ef7b4bc79d1a2e8cf2aa51e54dc1a5 for a full test report.

**Comment by akesandgren** [comment]
@boegel  Seems to work now!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3159.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/73f8920fbf45e560fd93f26b0db3cdb4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3407.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/bafedbe7ed813b4b2e4bd6c007b7d683 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2492.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/84b77ad6f00720a2d047414e3e2ed55e for a full test report.

**Comment by boegel** [comment]
Going in, thanks @boegel!

**Comment by boegel** [comment]
Oops, didn't mean to merge this myself, but @akesandgren was having trouble with `--merge-pr` and I was trying to help out. I guess I did.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 62 out of 64 (1 easyconfigs in this PR)
arm1 - Linux ubuntu 18.04, AArch64, UNKNOWN (aarch64), Python 3.6.9
See https://gist.github.com/7a43fc20b06e8ce566514ef238c78460 for a full test report.

**Pull request merged**